### PR TITLE
Feature/combine init and signup behavior

### LINKED ISSUE #76: Export deploy app script for relay test
**Is your feature request related to a problem? Please describe.**
In the current testing flow, we move the contract into relay/test for testing the relayer.

**Describe the solution you'd like**
Export the deploy app script in contract/ and import it inside the relay test.

**Additional context**
Learning from the style of Unirep Social. Thanks to @vivianjeng for suggesting this.

### LINKED ISSUE #78: [Login][Enhancement] Combine Login(Init) and SignUp
**Is your feature request related to a problem? Please describe.**
Currently, we separate two behaviors including loginOrInit and signUp.
Could we simplify these behaviors to improve user experience and reduce the transaction?

**Describe the solution you'd like**
Combine these two behaviors to one for easily signUp

**Describe alternatives you've considered**
Need to discuss further

0907 discussions 
1. login callback with OauthToken (need to discuss how to signUp with wallet)
2. Use DB (need to discuss how to extend relayer)
3.

### LINKED ISSUE #83: Contracts Test Broken
**Describe the bug**
The test of contracts fails.

**To Reproduce**

First, at the root dir of main branch, run
```
yarn install && yarn build
```
Which would build the files you needed for test.

After that, run
```
yarn contracts test
```

to run the tests of contracts.

The error should appear.

**Expected behavior**
All of the tests of contracts shall pass.

**Screenshots**
image

**Additional context**
Maybe this has something to do with #76.

### LINKED ISSUE #84: login ui update
**Is your feature request related to a problem? Please describe.**
Add user interface feature for user loggin but hasn't signed up

**Describe the solution you'd like**
Add user interface feature for user loggin but hasn't signed up

**Describe alternatives you've considered**
Match the design

### PR DESCRIPTION
**Pull request by chentihe**
## Summary

Remove status related code from the contract and combine initStatus and signUp into one function.
fix the timeout error of contract unit test.

## Linked Issue

close #78, close #76, close #83, close #84

## Details

- add handle sign up event to fetch the data into db
- remove status related code from the contract
- remove identity api
- query user status from db instead of contract

## Tests

- add a user state factory to create user state easily.

## Todo

-   [ ] modify login test, if we are able to get when the epoch is due
-   [ ] post test may occur the below error, should find a way to avoid this error
image

## Checklist

-   [x] remove identity api
-   [x] modify login test
-   [x] combine initStatus and signUp into one function
-   [x] fix the timeout error of contract unit test.

### CODE REVIEW

**Comment by badukwei** [comment]
I would like to add a logout button for the pr. Please wait for the commit.

**Comment by badukwei** [comment]
I have done my work.

**Review by badukwei — COMMENTED** [review]
I checked the codebase for login and sign up part and it seems ok. I am not sure about the posting, so please check it.

**Comment by badukwei** [comment]
This will also solvethe issue
https://github.com/social-tw/social-tw-website/issues/84

**Review by badukwei — COMMENTED** [review]
I have reviewed the fronted part for login changes. The codebase is fine. Backend, contract and test part need to be reviewed. @NOOMA-42 @iverJisty

File: packages/relay/test/login.test.ts [hunk 1]
```diff
@@ -134,40 +158,53 @@ describe('LOGIN /login', () => {
             })
     })
 
-    it('/api/identity, before wallet sign up and after init user', async () => {
-        const wallet = TransactionManager.wallet!!
-        const expectedSignMsg = await wallet.signMessage(hashMockUserId)
+    it('/api/signup, user not sign up with wrong proof and return error', async function () {
```
[the remark this answers is not in the document]
Comment (chentihe) [reply packages/relay/test/login.test.ts]: Thank you for the review, the issues were modified and tested.

**Review by ryanycw — APPROVED** [review]
LGTM! Thank you everyone for all the work and discussions.

**Pull request merged**